gems/sorcery/0.16/sorcery.rbs

```diff
@@ -0,0 +1,22 @@
+module Sorcery
+  module Model
+    def authenticates_with_sorcery!: () -> void
+
+    module ClassMethods[T]
+      def authenticate: (*String credentials) ?{ () -> untyped } -> (T | nil)
```

Comment:
How about using `instance` type instead of `T` type parameter?

> instance denotes the type of instance of the class. class is the singleton of the class.
>
> https://github.com/ruby/rbs/blob/master/docs/syntax.md#base-types